configure.ac

```diff
@@ -239,9 +220,48 @@ m4_version_prereq(
 
 AC_CHECK_HEADERS([execinfo.h])
 
+htop_have_iokit=no
+htop_have_libproc=no
 if test "$my_htop_platform" = darwin; then
    AC_CHECK_HEADERS([mach/mach_time.h])
    AC_CHECK_TYPES([thread_extended_info_data_t], [], [], [[#include <mach/thread_info.h>]])
+   AC_CHECK_HEADERS([net/if_types.h])
+   AC_CHECK_HEADERS([net/route.h])
+   AC_CHECK_MEMBERS([struct vm_statistics64.compressor_page_count], [], [], [[#include <mach/mach_host.h>]])
+   AC_CHECK_MEMBERS([struct vm_statistics64.external_page_count], [], [], [[#include <mach/mach_host.h>]])
+
+   AC_CHECK_HEADERS([libproc.h], [have_libproc_h=yes], [have_libproc_h=no])
+   if test "x$have_libproc_h" = xyes; then
+      AC_CHECK_LIB([proc], [proc_pidinfo], [htop_have_libproc=yes], [htop_have_libproc=no])
+      if test "x$htop_have_libproc" = xyes; then
+         AC_DEFINE([HAVE_LIBPROC], [1], [Define if libproc APIs are available.])
+      fi
+   fi
+
+   htop_have_iokit=yes
+   AC_CHECK_HEADER([IOKit/IOKitLib.h], [], [htop_have_iokit=no])
+   AC_CHECK_HEADER([IOKit/IOTypes.h], [], [htop_have_iokit=no])
+   AC_CHECK_HEADER([IOKit/ps/IOPowerSources.h], [], [htop_have_iokit=no])
+   AC_CHECK_HEADER([IOKit/ps/IOPSKeys.h], [], [htop_have_iokit=no])
+   AC_CHECK_HEADER([IOKit/storage/IOBlockStorageDriver.h], [], [htop_have_iokit=no])
+   if test "x$htop_have_iokit" = xyes; then
+      AC_MSG_CHECKING([for usable IOKit framework])
+      save_LIBS="$LIBS"
+      LIBS="-framework IOKit -framework CoreFoundation $LIBS"
+      AC_LINK_IFELSE(
+         [AC_LANG_PROGRAM([[#include <IOKit/IOKitLib.h>]], [[return IOServiceMatching("IOGPU") == NULL;]])],
+         [have_iokit_link=yes],
+         [have_iokit_link=no]
+      )
```

Comment:
Why do we need this?

Reply:
I deleted these lines. I'll resolve this conversation.